src/api/store/testimonial.py

```diff
@@ -242,6 +237,11 @@ def create_testimonial(self, context: Context, args) -> Tuple[dict, MassEnergize
 
       new_testimonial.save()
       
+      if is_published:
+          new_testimonial.published_at = parse_datetime_to_aware()
+          add_auto_shared_communities_to_testimonial(new_testimonial)
+      new_testimonial.save()
```

Comment:
In the code below, we update the `new_testimonial` variable and save it. But after setting the audience, shouldn’t we save it again to make sure the changes to `new_testimonial` are kept?. Thoughts cc @abdullai-t 

```
      new_testimonial.save()

      if audience:
        new_testimonial.audience.set(audience)
```

Reply:
The `audience` field in the `Testimonial` model is a ManyToMany relationship. Django automatically creates a bridge table to store data for such fields. Using `.set()` or `.add()` on a ManyToMany field, like in `new_testimonial.audience.set(audience)`, saves the related instances without requiring an explicit `.save()` call on the `Testimonial` instance.